engine/consensus/dkg/processor/engine.go

```diff
@@ -0,0 +1,195 @@
+package processor
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/dkg"
+	"github.com/onflow/flow-go/network"
+)
+
+// Engine implements the crypto DKGProcessor interface which provides means for
+// DKG nodes to exchange private and public messages. The same instance can be
+// used across epochs, but transitions should be accompanied by a call to
+// SetEpoch, such that DKGMessages can be prepended with the appropriate epoch
+// ID.
+type Engine struct {
+	unit        *engine.Unit
+	log         zerolog.Logger
+	me          module.Local
+	conduit     network.Conduit
+	msgCh       chan dkg.DKGMessage
+	committee   flow.IdentifierList
+	myIndex     int
+	epochID     flow.Identifier
+	epochIDLock sync.Mutex
+}
+
+// New returns a new DKGProcessor engine. Instances participanting in a common
+// DKG protocol should use the same _ordered_ list of committee identifiers,
+// because senders and recipients are identified by their index in this list.
+// The msgCh is used to forward incoming DKG messages to the consumer, such as
+// the Controller implemented in the DKG module.
+func New(
+	logger zerolog.Logger,
+	net module.Network,
+	me module.Local,
+	msgCh chan dkg.DKGMessage,
+	committee flow.IdentifierList,
+	epochID flow.Identifier) (*Engine, error) {
```

Comment:
Epochs within a chain are identified by a `uint64` counter. Does this `epochID` need to be unique across different possible chains?

Reply:
Thanks! Was waiting for you to confirm the type :)
No I don't think they need to be unique across chains.